pkg/storage/storage.go

```diff
@@ -588,11 +588,14 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
 		return errors.ErrRepoNotFound
 	}
 
+	isTag := false
+
 	// as per spec "reference" can only be a digest and not a tag
 	digest, err := godigest.Parse(reference)
 	if err != nil {
-		is.log.Error().Err(err).Msg("invalid reference")
-		return errors.ErrBadManifest
+		is.log.Debug().Str("invalid digest: ", reference).Msg("storage: assuming tag")
```

Comment:
Is there a good way to see if it's a tag?

should we at least check if it matches a pattern? If not well defined than something as simple as having characters around `:`

Reply:
we can enforce a check to look for specific set of tags following this rule https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules but as of now we do not set any restriction on tag during upload so if we want to enforce this check we should do same for upload also.